admin/src/bench/mod.rs

```diff
@@ -242,26 +262,55 @@ fn writer(
 		db.commit(commit.drain(..).map(|(k, v)| (0, k, v))).unwrap();
 		COMMITS.fetch_add(1, Ordering::Relaxed);
 		commit.clear();
+
+		num_commits_done += 1;
+
+		if num_commits_done % args.writer_commits_per_sleep == 0 && args.writer_sleep_time > 0 {
+			thread::sleep(std::time::Duration::from_millis(args.writer_sleep_time));
+		}
 	}
 }
 
-fn reader(db: Arc<Db>, pool: Arc<SizePool>, seed: u64, index: u64, shutdown: Arc<AtomicBool>) {
+fn reader(
+	db: Arc<Db>,
+	args: Arc<Args>,
+	pool: Arc<SizePool>,
+	seed: u64,
+	index: u64,
+	shutdown: Arc<AtomicBool>,
+) {
 	// Query random keys while writing
 	let mut rng = rand::rngs::SmallRng::seed_from_u64(seed + index);
 	while !shutdown.load(Ordering::Relaxed) {
 		let commits = COMMITS.load(Ordering::Relaxed) as u64;
 		if commits == 0 {
 			continue
 		}
-		let num_keys = commits * COMMIT_SIZE as u64;
-		let key = pool.key(rng.next_u64() % num_keys + seed);
-		match db.get(0, &key).unwrap() {
-			Some(_) => {
-				QUERIES_HIT.fetch_add(1, Ordering::SeqCst);
-			},
-			None => {
-				QUERIES_MISS.fetch_add(1, Ordering::SeqCst);
-			},
+		if args.archive || args.reader_check_pruned {
+			let num_keys = commits * COMMIT_SIZE as u64;
+			let key = pool.key(rng.next_u64() % num_keys + seed);
+			match db.get(0, &key).unwrap() {
+				Some(_) => {
+					QUERIES_HIT.fetch_add(1, Ordering::SeqCst);
+				},
+				None => {
+					QUERIES_MISS.fetch_add(1, Ordering::SeqCst);
+				},
+			}
+		} else {
+			let num_commit_values = (COMMIT_SIZE - COMMIT_PRUNE_SIZE) as u64;
+			let num_keys = commits * num_commit_values;
+			let mut index = rng.next_u64() % num_keys;
+			index += (index / num_commit_values + 1) * COMMIT_PRUNE_SIZE as u64;
+			let key = pool.key(index + seed);
+			match db.get(0, &key).unwrap() {
+				Some(_) => {
+					QUERIES_HIT.fetch_add(1, Ordering::SeqCst);
+				},
+				None => {
+					QUERIES_MISS.fetch_add(1, Ordering::SeqCst);
+				},
+			}
```

Comment:
To avoid a bit or repetition
```suggestion
		let key = if args.archive || args.reader_check_pruned {
			let num_keys = commits * COMMIT_SIZE as u64;
			pool.key(rng.next_u64() % num_keys + seed)
		} else {
			let num_commit_values = (COMMIT_SIZE - COMMIT_PRUNE_SIZE) as u64;
			let num_keys = commits * num_commit_values;
			let mut index = rng.next_u64() % num_keys;
			index += (index / num_commit_values + 1) * COMMIT_PRUNE_SIZE as u64;
			pool.key(index + seed)
		};
		match db.get(0, &key).unwrap() {
			Some(_) => {
				QUERIES_HIT.fetch_add(1, Ordering::SeqCst);				
			},
			None => {
				QUERIES_MISS.fetch_add(1, Ordering::SeqCst);
			},
		}
```

Reply:
Done